nucypher/blockchain/eth/interfaces.py

```diff
@@ -267,7 +272,7 @@ def get_contract_factory(self, contract_name: str) -> Contract:
         except KeyError:
             raise self.UnknownContract('{} is not a locally compiled contract.'.format(contract_name))
         except TypeError:
-            if self.__raw_contract_cache is constants.NO_COMPILATION_PERFORMED:
+            if self.__raw_contract_cache is NO_COMPILATION_PERFORMED:
```

Comment:
What if `TypeError` is raised for a different reason?

Reply:
:heavy_check_mark: